class.jetpack-gutenberg.php

```diff
@@ -99,39 +111,55 @@ protected static function share_items( $a, $b ) {
 		return count( array_intersect( $a, $b ) ) > 0;
 	}
 
+	/**
+	 * Register an extension
+	 *
+	 * If the extension isn't whitelisted, set its unavailability reason instead.
+	 *
+	 * @param string $slug Slug of extension block.
+	 *
+	 * @return boolean True if the extension is registered and False if it is not whitelisted
+	 */
+	public static function register_extension( $slug ) {
+		if ( ! self::is_extension_whitelisted( $slug ) ) {
+			self::set_extension_unavailability_reason( $slug, 'not_whitelisted' );
+			return false;
+		}
+
+		self::$registered_extensions[] = $slug;
+		return true;
+	}
+
 	/**
 	 * Register a block
 	 *
 	 * If the block isn't whitelisted, set its unavailability reason instead.
 	 *
+	 * @deprecated 7.0.0 Use register_extension() + register_block_type() instead
+	 *
 	 * @param string $slug Slug of the block.
 	 * @param array  $args Arguments that are passed into register_block_type().
 	 */
 	public static function register_block( $slug, $args ) {
 		$prefixed_extensions = array_map( array( __CLASS__, 'prepend_block_prefix' ), self::$extensions );
 
 		// Register the block if it's whitelisted, or if it's a child block, and one of its parents is whitelisted.
-		if ( in_array( $slug, self::$extensions, true ) || ( isset( $args['parent'] ) && self::share_items( $args['parent'], $prefixed_extensions ) ) ) {
+		if ( self::register_extension( $slug ) || ( isset( $args['parent'] ) && self::share_items( $args['parent'], $prefixed_extensions ) ) ) {
```

Comment:
Does it matter that an extension can potentially be registered ( added to self::$registered_extensions[]) multiple times?

Reply:
That's true. In fact `register_block_type` would raise an error if the we previously registered a block with the same name, so I included an additional check in 80b379a.